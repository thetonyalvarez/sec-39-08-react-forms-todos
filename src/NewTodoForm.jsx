import { useState } from 'react'
import { useFormik } from 'formik'
import * as yup from 'yup';
import Stack from '@mui/material/Stack'
import Button from '@mui/material/Button'
import TextField from '@mui/material/TextField'
import Grid from '@mui/material/Grid'
import './NewTodoForm.css'

// set up form validation
const validationSchema = yup.object({
  task: yup
    .string('Task cannot be empty.')
    .required('Required'),
});
// end form validation


const NewTodoForm = ({ addTodo }) => {

  const INITIAL_STATE = { task: "" }
  const [formData, setFormData] = useState(INITIAL_STATE)

  // send bgColor, width, and height to parent form and clear form
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo(formData);
    setFormData(INITIAL_STATE);
  };

  const handleChange = (e) => {
    const { id, value } = e.target;
    setFormData(fData => ({
      ...fData,
      [id]: value
    })) 
  }

  const formik = useFormik({
    initialValues: {
      task: ""
    },
    validationSchema: validationSchema,
    onSubmit: (values, { resetForm }) => {
      addTodo(values);
      resetForm({
        task: ""
      })
    }
  })


  return (
    <Grid item xs={12} md={6}>
      <Stack
        justifyContent="center"
        component="form"
        autoComplete="off"
        sx={{ mb: 4 }}
        onSubmit={formik.handleSubmit}
      >
        <TextField 
          id="task"
          label="Task"
          variant="standard"
          sx={{ mb: 2 }}
          value={formik.values.task}
          onChange={formik.handleChange}
          error={formik.touched.task && Boolean(formik.errors.task)}
          helperText={formik.touched.task && formik.errors.task}
        />

        <Button variant="contained" color="secondary" type="submit">Add new task</Button>

      </Stack>
    </Grid>
  )
}

export default NewTodoForm;