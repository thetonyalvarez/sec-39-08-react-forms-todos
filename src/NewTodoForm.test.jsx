import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import NewTodoForm from './NewTodoForm'

afterEach(cleanup)

// smoke test
it("renders without crashing", function() {
  render(<NewTodoForm />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
