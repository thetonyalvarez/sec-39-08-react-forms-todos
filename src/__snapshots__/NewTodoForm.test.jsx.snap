// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-13i4rnv-MuiGrid-root"
  >
    <form
      autocomplete="off"
      class="css-ow1s07-MuiStack-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1sqs9ir-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="task"
          id="task-label"
        >
          Task
        </label>
        <div
          class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1ptx2yq-MuiInputBase-root-MuiInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
            id="task"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Add new task
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
